hydra-node/src/Hydra/Chain.hs

```diff
@@ -134,16 +134,8 @@ data PostTxError tx
   | -- | Comitting reference scripts is not supported right now.
     CannotCommitReferenceScript
   | InvalidStateToPost {txTried :: PostChainTx tx, chainState :: ChainStateType tx}
-  | NotEnoughFuel
-  | NoFuelUTXOFound
-  | -- | Script execution failed when finalizing a transaction in the wallet.
-    -- XXX: Ideally we want a cardano-api type with corresonding JSON instance
-    -- here. But the wallet still uses ledger types and we don't want to copy the
-    -- conversion from ledger 'TransactionScriptFailure' to the cardano-api
-    -- 'ScriptExecutionError' type.
-    ScriptFailedInWallet {redeemerPtr :: Text, failureReason :: Text}
```

Comment:
Why did you remove these errors?

Especially the errors for `NotEnoughFuel` or `NoFuelUTxOFound` are meaningful to the clients and require user action!

Also, the removal of `ScriptFailedInWallet` seems very unrelated to the goal of this PR

Reply:
I've removed them since I wanted to get rid of `PostTxError` being a kitchen sink for different kinds of errors. Let's design a better solution then, perhaps in another PR.